### ISSUE TITLE
Undefined reference opal_hwloc

### ISSUE THREAD

**Issue by solalito** [open-mpi/ompi#6907]
Thank you for taking the time to submit an issue!

## Background information
Running a simple C Hello World program fails with mpicc, yielding an error about undefined reference to a number of opal functions.

### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)
Open MPI 3.1.0

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
Installed it from source code.
./configure --prefix=$LOCAL_DIR CC=PATH_TO_MY_CC FC=PATH_TO__MY_FC CXX=PATH_TO_MY_CXX --disable-multilib 

### Please describe the system on which you are running

* Operating system/version: Ubuntu 18.04.3 (as a subsystem on Windows 10)
* Computer hardware: Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
* Network type: None

-----------------------------

## Details of the problem

I am testing my newly installed OpenMPI compilers on the "Hello World" C code you provide on Git: https://github.com/open-mpi/ompi/blob/master/examples/hello_c.c

Running this yields the following error:

```shell
/home/samouyal/usr/lib/libmpi.so: undefined reference to `opal_hwloc1117_hwloc_bitmap_iszero'
/home/samouyal/usr/lib/libmpi.so: undefined reference to `opal_hwloc1117_hwloc_bitmap_free'
/home/samouyal/usr/lib/libmpi.so: undefined reference to `opal_hwloc1117_hwloc_get_cpubind'
/home/samouyal/usr/lib/libmpi.so: undefined reference to `opal_hwloc1117_hwloc_get_obj_by_depth'
/home/samouyal/usr/lib/libmpi.so: undefined reference to `opal_hwloc1117_hwloc_bitmap_isincluded'
/home/samouyal/usr/lib/libmpi.so: undefined reference to `opal_hwloc1117_hwloc_bitmap_compare'
/home/samouyal/usr/lib/libmpi.so: undefined reference to `opal_hwloc1117_hwloc_get_type_depth'
/home/samouyal/usr/lib/libmpi.so: undefined reference to `opal_hwloc1117_hwloc_bitmap_alloc'
```

I've found a similar issue #3763 which was closed due to inactivity but remained unresolved. Output to the follow-up questions:

```shell
        ldd /home/samouyal/usr/lib/libopen-rte.so.12

        linux-vdso.so.1 (0x00007fffcf362000)
        libopen-pal.so.40 => /home/samouyal/usr/lib/libopen-pal.so.40 (0x00007f86924f0000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f86922e0000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f86920d0000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f8691d30000)
        libutil.so.1 => /lib/x86_64-linux-gnu/libutil.so.1 (0x00007f8691b20000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f8691900000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f86914f0000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f8692c00000)
```
libopen-pal.so.40 is in the same directory.

```shell
        nm /home/samouyal/lib/libopen-pal.so.* | grep opal_hwloc1117_hwloc_bitmap_iszero

        000000000008bb10 T opal_hwloc1117_hwloc_bitmap_iszero
```
Therefore opal_hwloc1117_hwloc_bitmap_iszero seems to be linked correctly.

Thank you for the help

**Comment by jsquyres** [comment]
Can you both upgrade to the latest Open MPI 3.1.x (i.e., 3.1.4) and double check that your `$LD_LIBRARY_PATH` points to the correct Open MPI installation?

**Comment by ggouaillardet** [comment]
Also you want to run `ldd hello` before investigating your libraries installed in `/home/samouyal/usr/lib/`

**Comment by solalito** [comment]
I installed Open MPI 3.1.4 and got the same error. Note that I've made a mistake in my original post and that the above error happens at compile-time, not run-time. 

After installing the new version of OpenMPI, I noticed that not all of the libraries in the `/home/samouyal/usr/lib` had been overwritten (e.g. `libopen-pal.so.x.y.z). Also note that other versions of OpenMPI had been installed in the same directory. I thought perhaps `mpicc`  was reading the wrong libraries belonging to other versions yielding the linking errors. Therefore, I removed all MPI related libraries (libmpi, liboshmem, etc...) and reinstalled them with `make install`. That fixed the isssue.

Thanks for your help!
